File: pr1-java/services/src/main/java/pr1Java/services/IServices.java
package pr1Java.services;

import pr1Java.model.Game;
import pr1Java.model.User;
import pr1Java.model.observers.IObserver;

import java.util.Collection;

public interface IServices {
    User signInUser(String username, String password, IObserver client) throws Exception;

    void signOutUser(String username, IObserver client) throws Exception;

    User signUpUser(String username, String password, IObserver client) throws Exception;

    Collection<Game> getAllGames() throws Exception;

    void sellSeats(Game game, String clientName, Integer seatsCount) throws Exception;

    Collection<Game> getGamesWithAvailableSeatsDescending() throws Exception;
}

File: pr1-cs/Services/IServices.cs
using System.Collections.Generic;
using Model;
using Model.Observers;

namespace Services
{
    public interface IServices
    {
        User SignInUser(string username, string password, IObserver client);

        void SignOutUser(string username, IObserver client);

        User SignUpUser(string username, string password, IObserver client);

        IEnumerable<Game> GetAllGames();

        void SellSeats(Game game, string clientName, int seatsCount);

        IEnumerable<Game> GetGamesWithAvailableSeatsDescending();
    }
}